### PR TITLE
implement generator in Scheme

### PR DESCRIPTION
**Pull request by burgerrg**
Remove the complex dependencies that result from C++ code now that protobuf requires third-party libraries.
Tested against protobuf v24.2.

### CODE REVIEW

File: generator/generator.ss [hunk 1]
```diff
@@ -0,0 +1,142 @@
+(library (generator generator)
```
Comment (wintonpc) [inline generator/generator.ss:18]: Preserve the copyright and license from `scheme_generator.cc`?

File: protoc-gen-scheme [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env swish
```
Comment (wintonpc) [inline protoc-gen-scheme:1]: What's the plan to make this work in web? The web docker image doesn't have swish installed. (And neither does my machine.)

File: protoc-gen-scheme [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+(let ([op (console-error-port)])
+  ;; Don't send anything to stdout except the response message
+  (current-output-port op)
+  (console-output-port op)
+  (trace-output-port op))
+(let ([dir (path-parent (car (command-line)))])
+  (library-directories (list (cons dir dir))))
```
Comment (wintonpc) [inline protoc-gen-scheme:8]: In web, we'll need to keep the old `library-directories` that are set up by `setup.ss` so we can `(import (swish imports))` from swish-lite.

**Review by wintonpc — CHANGES_REQUESTED** [review]
Bootstrapping was a great insight!

File: generator/generator.ss (hunk 1 above)
Comment (burgerrg) [reply generator/generator.ss:18]: Good idea, will do!

File: protoc-gen-scheme (hunk 2 above)
Comment (burgerrg) [reply protoc-gen-scheme:1]: See https://github.com/indigobio/ascent-web/blob/protobuf/referee/src/protoc-gen-scheme.in

File: protoc-gen-scheme (hunk 3 above)
Comment (burgerrg) [reply protoc-gen-scheme:8]: Yes, so we'll not use this protoc-gen-scheme at all.

**Review by wintonpc — APPROVED** [review]

**Pull request merged**